### PR TITLE
[Fix] `newline-after-import`: recognize decorators

### PR DESCRIPTION
**Pull request by atos1990**
Hi, this is my fix issue https://github.com/benmosher/eslint-plugin-import/issues/1004

### CODE REVIEW

**Comment by atos1990** [comment]
@ljharb, done

File: tests/src/rules/newline-after-import.js [hunk 1]
```diff
@@ -164,6 +164,24 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
       parserOptions: { sourceType: 'module' },
       parser: 'babel-eslint',
     },
+    {
+      code : `// issue 1004
+        import foo from 'foo';\n
+        @SomeDecorator(foo)
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js]: hm, if a newline is required after `import`, this should be invalid - there needs to be a blank line between the `import` line and the decorator.

File: tests/src/rules/newline-after-import.js [hunk 2]
```diff
@@ -164,6 +164,24 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
...
+        export default class Test {}
+      `,
+      parserOptions: { sourceType: 'module' },
+      parser: 'babel-eslint',
+    },
+    {
+      code : `// issue 1004
+        const foo = require('foo');\n
+        @SomeDecorator(foo)
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js]: same here

File: tests/src/rules/newline-after-import.js [hunk 3]
```diff
@@ -349,5 +367,43 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
...
     },
+    {
+      code: `// issue 10042
+        import foo from 'foo';
+        @SomeDecorator(foo)
+        export default class Test {}
+      `,
+      output: `// issue 10042
+        import foo from 'foo';\n
+        @SomeDecorator(foo)
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js]: this should autofix to add a blank line above the decorator

File: tests/src/rules/newline-after-import.js [hunk 4]
```diff
@@ -349,5 +367,43 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
...
+    },
+    {
+      code: `// issue 1004
+        const foo = require('foo');
+        @SomeDecorator(foo)
+        export default class Test {}
+      `,
+      output: `// issue 1004
+        const foo = require('foo');\n
+        @SomeDecorator(foo)
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js]: this should autofix to add a blank line above the decorator

File: tests/src/rules/newline-after-import.js (hunk 1 above)
Comment (atos1990) [reply tests/src/rules/newline-after-import.js]: But at the end of line 169 - '\n', i.e. blank line. Is it not enough?
I think that in this case it is correct, isn't it?

File: tests/src/rules/newline-after-import.js [hunk 5]
```diff
@@ -349,5 +359,27 @@ ruleTester.run('newline-after-import', require('rules/newline-after-import'), {
       parserOptions: { sourceType: 'module' },
       parser: 'babel-eslint',
     },
+    {
+      code: `// issue 10042\nimport foo from 'foo';\n@SomeDecorator(foo)\nexport default class Test {}`,
```
Comment (ljharb) [inline tests/src/rules/newline-after-import.js]: can these be reverted to use actual newlines instead of `\n`? it's much more readable.

**Comment by ljharb** [comment]
ping @atos1990, this also needs a rebase

**Comment by TrejGun** [comment]
@atos1990, please rebase

**Comment by shastaxc** [comment]
Came here looking for this fix. ETA on merge? With this rule not working with decorators, it pretty much means it won't work with any of my ts files in Angular, which is the only thing I'm using ESLint for.

**Comment by TrejGun** [comment]
thanks @ljharb

**Comment by elbertcastaneda** [comment]
We continue with this error in the latest version, I am thinking that the next version could contain this fix, is it right?

**Comment by ljharb** [comment]
Yes.

**Comment by ljharb** [comment]
@skoblenick as you can see if you click on the [commit that closed the PR](https://github.com/import-js/eslint-plugin-import/commit/b307c7c14bed8b20807cc629911093c5ec47f0ed), it is in both main, and every version v2.21.0 and later.

**Pull request merged**